rest/src/main/java/discord4j/rest/http/client/ClientException.java

```diff
@@ -173,32 +172,6 @@ public static Predicate<Throwable> isStatusCode(Integer... codes) {
         };
     }
 
-    /**
-     * {@link Predicate} helper to further classify a {@link ClientException}, while creating a {@link Retry}
-     * factory, depending on the underlying HTTP status code. A {@link Retry} factory can be created through methods
-     * like {@link reactor.retry.Retry#onlyIf(Predicate)} where this method can be used as argument.
-     *
-     * @param code the status code for which this {@link Predicate} should return {@code true}
-     * @return a {@link Predicate} that returns {@code true} if the given {@link RetryContext} exception is a
-     * {@link ClientException} containing the given HTTP status code
-     */
-    public static Predicate<RetryContext<?>> isRetryContextStatusCode(int code) {
-        return ctx -> isStatusCode(code).test(ctx.exception());
-    }
-
-    /**
-     * {@link Predicate} helper to further classify a {@link ClientException}, while creating a {@link Retry}
-     * factory, depending on the underlying HTTP status code. A {@link Retry} factory can be created through methods
-     * like {@link reactor.retry.Retry#onlyIf(Predicate)} where this method can be used as argument.
-     *
-     * @param codes the status codes for which this {@link Predicate} should return {@code true}
-     * @return a {@link Predicate} that returns {@code true} if the given {@link Throwable} is a {@link ClientException}
-     * containing the given HTTP status code
-     */
-    public static Predicate<RetryContext<?>> isRetryContextStatusCode(Integer... codes) {
-        return ctx -> isStatusCode(codes).test(ctx.exception());
-    }
-
```

Comment:
The new Retry usage lets us perform directly those functions, I removed those functions are they are only used with the legacy implementaition.